src/main/java/com/cloudbees/jenkins/support/filter/DefaultContentFilter.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.cloudbees.jenkins.support.filter;
+
+import com.cloudbees.jenkins.support.SupportPlugin;
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.Functions;
+import hudson.XmlFile;
+import hudson.model.AbstractItem;
+import hudson.model.ManagementLink;
+import hudson.model.Saveable;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.kohsuke.randname.RandomNameGenerator;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+import org.slf4j.Logger;
```

Comment:
I think you should use `java.util.logging` for consistency.

Reply:
I'd prefer to use log4j-api, but it's not a dependency yet, and I'd rather discuss a larger JEP or whatever to address a mass migration. As for this use case, well, I'll figure out how to refactor to that.

Reply:
To me, the user value will be that Jenkins and plugins themselves will be able to more easily log things (with more contextual info) that can make troubleshooting and debugging easier. However, I think I'll need to get more familiar with Jenkins at large to devise a proper proposal.